classes/core/PKPEventServiceProvider.inc.php

```diff
@@ -27,7 +38,23 @@ class PKPEventServiceProvider extends EventServiceProvider
      * @var array $listen $event => $listeners[]
      * @brief Registering events & listeners, see Illuminate\Events\EventServiceProvider
      */
-    protected $listen = [];
+    protected $listen = [
```

Comment:
I can imagine that this declaration will grow really long if we adopt this practice throughout the software. I'm guessing there are other ways to associate events with listeners?

Reply:
We could move to use [`Event Subscribers`](https://laravel.com/docs/7.x/events#event-subscribers) entirely.

Reply:
Those subscriptions would need to be scanned and cached, correct?